src/app/predict/uploadMedia/page.tsx

```diff
@@ -0,0 +1,113 @@
+"use client";
+
+import PredictionLayout from "@/components/PredictionLayout";
+import SquareIconButton from "@/components/SquareIconButton";
+import { Model } from "@/types/model";
+import { useRef, useState } from "react";
+interface MediaState {
+  typeOfMedia: "video" | "image" | null;
+}
+
+export default function RealtimePredictionPage() {
+  const [mediaUpload, setMediaUpload] = useState<boolean>(false);
+  const [mediaState, setMediaState] = useState<MediaState>({
+    typeOfMedia: null,
+  });
+  const mediaRef = useRef<HTMLMediaElement>(null);
+  const [mediaList, setMediaList] = useState<string[]>([]); // URLs of the uploaded media
+  const [selectedModel, setSelectedModel] = useState<Model | undefined>(
+    undefined
+  );
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const files = event.target.files;
+    if (!files || files.length === 0) return; // No file selected
+
+    const newMediaList: string[] = [];
+    for (let i = 0; i < files.length; i++) {
+      const file = files[i];
+      const reader = new FileReader();
+
+      reader.onload = () => {
+        const url = reader.result as string;
+        newMediaList.push(url);
+
+        if (i === files.length - 1) {
+          // All files have been processed, update state
+          setMediaList([...mediaList, ...newMediaList]);
+          setMediaUpload(true);
+        }
+      };
+
+      if (file.type.startsWith("image/")) {
+        reader.readAsDataURL(file);
+        setMediaState({ typeOfMedia: "image" });
+      } else if (file.type.startsWith("video/")) {
+        reader.readAsDataURL(file);
+        setMediaState({ typeOfMedia: "video" });
+      } else {
+        console.error("Unsupported file type:", file.type);
+      }
+    }
+  };
+
+  // Function to trigger file input click
+  const handleButtonClick = () => {
+    document.getElementById("fileInput")?.click();
+  };
+
+  return (
+    <PredictionLayout
+      inputDescriptionText={
+        !mediaUpload
+          ? "Upload Media"
+          : mediaList.length > 0 && mediaList[0].startsWith("data:image")
+          ? "Images"
+          : "Video"
```

Comment:
Why only look at the first file type? What if there are both images and videos selected. May fix it to be a single word "Selected Media".